### PR TITLE
support install external agent in Sermant

### LINKED ISSUE #1705: Support install Otel agent in Sermant
### What would you like to be added?

Support install external agent(like Otel agent) in Sermant

### Why is this needed?

To better integrate with OpenTelemetry and make it easier for users to leverage its observability capabilities, we aim to treat OpenTelemetry as a component within Sermant. It will support both static and dynamic attachment methods. Furthermore, we decide to support any external agent in Sermant

### PR DESCRIPTION
**Pull request by lilai23**
**What type of PR is this?**

Feature

**What this PR does / why we need it?**

To better integrate with OpenTelemetry and make it easier for users to leverage its observability capabilities, we aim to treat OpenTelemetry as a component within Sermant. It will support both static and dynamic attachment methods.

**Which issue(s) this PR fixes？**

Fixes #1705

**Does this PR introduce a user-facing change?**

Yes. We will update user guide at sermant.io.

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/sermant-io/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [ ] GitHub Actions works fine in this PR.

### CODE REVIEW

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+     * @param agentPath OpenTelemetry agent file path
+     * @param argsMap arguments of the installation
+     * @param instrumentation instrumentation
+     * @throws IOException
+     * @throws NoSuchMethodException
+     * @throws ClassNotFoundException
+     * @throws InvocationTargetException
+     * @throws IllegalAccessException
+     */
+    static void installOtelAgent(boolean isDynamic, String agentPath, Map<String, String> argsMap,
```
Comment (zwmagic) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java]: The code for using installCommonAgent() and installOtelAgent() is duplicated, which can be merged

**Review by zwmagic — DISMISSED** [review]

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java [hunk 2]
```diff
@@ -136,6 +139,20 @@ public static void install(String artifact, Map<String, Object> argsMap, Instrum
         if (NotificationManager.isEnable()) {
             NotificationManager.doNotify(new NotificationInfo(SermantNotificationType.LOAD_COMPLETE, null));
         }
+
+        // cache instrumentation
+        CommandProcessor.cacheInstrumentation(instrumentation);
+
+        // install OpenTelemetry Agent
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java]: install external Agent，such as OpenTelemetry

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java [hunk 3]
```diff
@@ -136,6 +139,20 @@ public static void install(String artifact, Map<String, Object> argsMap, Instrum
...
+        AgentConfig agentConfig = ConfigManager.getConfig(AgentConfig.class);
+        if (agentConfig.isExternalAgentInjection()) {
+            try {
+                ExternalAgentManager.handleAgentInstallation(false, agentConfig.getExternalAgentName(),
+                        agentConfig.getExternalAgentFile(), null, instrumentation);
+            } catch (Exception e) {
+                LOGGER.severe("Failed to install external agent: " + e.getMessage());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java]: extract as method

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java [hunk 4]
```diff
@@ -136,6 +139,20 @@ public static void install(String artifact, Map<String, Object> argsMap, Instrum
         if (NotificationManager.isEnable()) {
             NotificationManager.doNotify(new NotificationInfo(SermantNotificationType.LOAD_COMPLETE, null));
         }
+
+        // cache instrumentation
+        CommandProcessor.cacheInstrumentation(instrumentation);
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java:144]: i think ExternalAgentManager caching instrumentation is better, and handleAgentInstallation method can reduce an input parameter

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/ExternalAgentInstallCommandExecutor.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+            LOGGER.severe("Failed to install external agent: AGENT_FILE in command args is empty");
+            return;
+        }
+
+        try {
+            ExternalAgentManager.handleAgentInstallation(true, args, agentPath, agentArgsMap,
+                    CommandProcessor.getInstrumentation());
+        } catch (IOException | NoSuchMethodException | ClassNotFoundException | InvocationTargetException
+                | IllegalAccessException e) {
+            LOGGER.severe("Failed to install external agent: " + e.getMessage());
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/ExternalAgentInstallCommandExecutor.java]: Failed to install external agent[agent name : args]:

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+     * @throws IOException
+     */
+    static String loadAgentJar(String agentName, String agentPath, Instrumentation instrumentation)
+            throws IOException {
+        File agentJarFile = new File(agentPath);
+        if (!agentJarFile.isFile()) {
+            throw new SermantRuntimeException("Invalid Jar file: " + agentJarFile);
+        }
+
+        String otelAgentClassName;
```
Comment (daizhenyu) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java]: externalAgentClassName

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java (hunk 4 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java:144]: instrumentation can be use by the second sermant agent install or uninstall in the future，so put it in CommandProcessor

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java (hunk 1 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java]: resolved as one method

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java (hunk 2 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java (hunk 3 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/AgentCoreEntrance.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/ExternalAgentInstallCommandExecutor.java (hunk 5 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/command/ExternalAgentInstallCommandExecutor.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java (hunk 6 above)
Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/ext/ExternalAgentManager.java]: done

File: sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/event/collector/FrameworkEventCollector.java [hunk 7]
```diff
@@ -164,4 +164,20 @@ public void collectdHotPluggingEvent(FrameworkEventDefinitions frameworkEventDef
                 frameworkEventDefinitions.getEventType(),
                 new EventInfo(frameworkEventDefinitions.getName(), description)));
     }
+
+    /**
+     * Collect OpenTelemetry Agent start event
+     *
+     * @param startMethod the method name OpenTelemetry Agent starts by
+     */
+    public void collectOtelStartEvent(String startMethod) {
```
Comment (zwmagic) [inline sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/event/collector/FrameworkEventCollector.java:173]: make this method more general and decouple it from Otel

Comment (lilai23) [reply sermant-agentcore/sermant-agentcore-core/src/main/java/io/sermant/core/event/collector/FrameworkEventCollector.java:173]: I think other agent start event are not sopported in Sermant. Once If a  specifc agent needs to be added into here.  We just add new interceptor and event

**Review by daizhenyu — APPROVED** [review]

**Review by hanbingleixue — APPROVED** [review]

**Pull request merged**